tests/delivery/test_deliveries.py

```diff
@@ -1,5 +1,227 @@
+        'message': msg,
+        'signature': w3.toHex(signed.signature),
+        'public_key': buyer
+    })
+    payload = json.loads(login.data)
+    access_token = payload['access_token']
+
+    # Get the listing
+    headers = {
+        'Authorization': f'Bearer {access_token}'
+    }
+    delivery = test_client.get(
+        '/deliveries/',
+        query_string={
+            'delivery_hash': w3.toHex(delivery_hash),
+            'listing_hash': w3.toHex(listing_hash)
+        },
+        headers=headers
+    )
+    assert delivery.status_code == 412
+
+def test_successful_delivery(w3, datatrust, pk, user, dynamo_table, s3_client, test_client):
+    # Add the listing to dynamo
+    buyer = w3.eth.accounts[10]
+    listing_hash = w3.keccak(text='a_witch')
+    file_contents = 'burn_the_witch'
+    mimetype = 'application/text'
+    row = {
+            'listing_hash': w3.toHex(listing_hash),
+            'title': 'so many catz',
+            'file_type': mimetype
+        }
+
+    g.table.put_item(Item=row)
+
+    # Put the listing in S3
+    s3_object = g.s3.put_object(
+        Body=file_contents,
+        Bucket=current_app.config['S3_DESTINATION'],
+        Key=w3.toHex(listing_hash)
+    )
+    
+    # Request a delivery
+    delivery_hash = w3.keccak(text='monty_pythons_delivery_hash')
+    amount = 14 # derived from storing the actual file in S3 and checking size
+    delivery_tx = transact(
+        datatrust.request_delivery(
+            delivery_hash, amount, 
+            {'from': buyer, 'gas': 1000000, 'gasPrice': w3.toWei(2, 'gwei')}
+        )
+    )
+    w3.geth.personal.lockAccount(buyer)
+    delivery_rct = w3.eth.waitForTransactionReceipt(delivery_tx)
+    dlvry = call(datatrust.get_delivery(delivery_hash))
+    assert dlvry[0] == buyer
+
+    # Get a jwt
+    msg = 'weighs as much as a duck'
+    signed = w3.eth.account.sign_message(encode_defunct(text=msg), private_key=pk)
+    login = test_client.post('/authorize/', json={
+        'message': msg,
+        'signature': w3.toHex(signed.signature),
+        'public_key': buyer
+    })
+    payload = json.loads(login.data)
+    access_token = payload['access_token']
+
+    # Get the listing
+    headers = {
+        'Authorization': f'Bearer {access_token}'
+    }
+    delivery = test_client.get(
+        '/deliveries/',
+        query_string={
+            'delivery_hash': w3.toHex(delivery_hash),
+            'listing_hash': w3.toHex(listing_hash)
+        },
+        headers=headers
+    )
+    #TODO: Assert delivered in protocol by checking the bytes_accessed for the listing
+    bytes_accessed = call(datatrust.get_bytes_accessed(listing_hash))
+    assert bytes_accessed == amount
+
+    # Assert the listing contents
+    mimetype = 'application/text'
+    assert delivery.headers['Content-Type'] == mimetype
+    assert int(delivery.headers['Content-Length']) == amount
+    assert delivery.data == file_contents.encode()
```

Comment:
check and assert in here that the <DatatrustAddress> got paid some ethertoken. seems like something they'd want to know